packages/js-client/src/context.ts

```diff
@@ -0,0 +1,71 @@
+import { SimpleStorageContextParams } from './types';
+import { Context, ContextCore, ContextState } from '@aragon/sdk-client-common';
+
+// TODO
+// import from @aragon/sdk-client-common
+type OverriddenState = {
+  daoFactoryAddress: boolean;
+  ensRegistryAddress: boolean;
+  gasFeeEstimationFactor: boolean;
+  ipfsNodes: boolean;
+  graphqlNodes: boolean;
+};
+
+type SimpleStorageContextState = ContextState & {
+  // extend the Context state with a new state for storing
+  // the new parameters
+  myParam: string;
+};
+
+type SimpleStorageOverridenState = OverriddenState & {
+  myParam: boolean;
+};
+
+export class SimpleStorageContext extends ContextCore {
+  // super is called before the properties are initialized
+  // so we initialize them to the value of the parent class
+  protected state: SimpleStorageContextState = this.state;
+  protected overriden: SimpleStorageOverridenState = this.overriden;
+  constructor(
+    contextParams?: Partial<SimpleStorageContextParams>,
+    context?: Context
+  ) {
+    // parent contructor will call this.set(contextParams)
+    // so we don't need to call it again
+    super(contextParams);
+    if (context) {
+      // copy the context properties to this
+      Object.assign(this, context);
+    }
+    if (contextParams) {
+      // overide the context params with the ones passed to the constructor
+      this.set(contextParams ?? {});
+    }
```

Comment:
That might be confusing for a newcomer. I would make the difference clear with something like:
1. `aragonContext`
2. `pluginContext`

Reply:
not sure what do you mean, rename context to `aragonContext` ?

Reply:
Having the custom settings of the plugin on an object called `pluginContext` and the general settings from Aragon's `Context` called `aragonContext` or `globalContext`